### PR TITLE
Sub mode support

### PR DESCRIPTION
**Pull request by AndreasAntener**
Updated variant of #3275

HIL tested

### CODE REVIEW

File: src/modules/commander/commander.cpp [hunk 1]
```diff
@@ -563,7 +564,27 @@ bool handle_command(struct vehicle_status_s *status_local, const struct safety_s
...
+						switch(custom_sub_mode) {
+						case PX4_CUSTOM_SUB_MODE_AUTO_LOITER:
+							main_ret = main_state_transition(status_local, vehicle_status_s::MAIN_STATE_AUTO_LOITER);
+							break;
+						case PX4_CUSTOM_SUB_MODE_AUTO_MISSION:
+							main_ret = main_state_transition(status_local, vehicle_status_s::MAIN_STATE_AUTO_MISSION);
+							break;
+						case PX4_CUSTOM_SUB_MODE_AUTO_RTL:
+							main_ret = main_state_transition(status_local, vehicle_status_s::MAIN_STATE_AUTO_RTL);
+							break;
```
Comment (LorenzMeier) [inline src/modules/commander/commander.cpp]: Can you please add AUTO_TAKEOFF here? I added that on master (also in QGC already).

Comment (AndreasAntener) [inline src/modules/commander/commander.cpp]: Yup, will rebase.

**Comment by LorenzMeier** [comment]
Awesome, thanks!

**Comment by LorenzMeier** [comment]
Might need another quick rebase on master.

**Comment by AndreasAntener** [comment]
Done

**Pull request merged**